crypto/engine/eng_devcrypto.c

```diff
@@ -598,9 +598,12 @@ static int devcrypto_digests(ENGINE *e, const EVP_MD **digest,
 static int devcrypto_unload(ENGINE *e)
 {
     destroy_all_cipher_methods();
-#if defined(COP_FLAG_UPDATE) && defined(COP_FLAG_FINAL)
+#ifdef IMPLEMENT_DIGEST
     destroy_all_digest_methods();
 #endif
+
+    close(cfd);
```

Comment:
how can you be sure that cfd is initialized,
given that engine_load_devcrypto_int() returns void?

Reply:
The `ENGINE_add` call at the end of `engine_load_devcrypto_int` makes the difference.

Reply:
Okay, but then we have a resource leak if the execution does not reach that point.

Reply:
Huh? Could you elaborate?

Reply:
I think I saw what you see.  Commit coming up!